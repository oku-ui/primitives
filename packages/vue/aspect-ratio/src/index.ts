export {
  AspectRatio,
  OkuAspectRatio,
} from './aspect-ratio'
export type { AspectRatioProps } from './aspect-ratio'
