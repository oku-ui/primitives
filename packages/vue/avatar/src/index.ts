export {
  OkuAvatar,
  OkuAvatarFallback,
  OkuAvatarImage,
  createAvatarScope,
} from './avatar'
export type { AvatarFallbackProps, AvatarProps, AvatarImageProps } from './avatar'
