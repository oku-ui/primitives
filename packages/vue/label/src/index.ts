export {
  OkuLabel,
  label,
} from './label'
export type { LabelProps } from './label'
