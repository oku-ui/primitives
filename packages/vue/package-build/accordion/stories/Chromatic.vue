<script setup lang="ts">
import { ref } from 'vue'
import { OkuAccordion, OkuAccordionContent, OkuAccordionHeader, OkuAccordionItem, OkuAccordionTrigger } from '@oku-ui/accordion'

const values = ref(['One', 'Two', 'Three', 'Four'])
</script>

<template>
  <div>
    <h1>Uncontrolled</h1>
    <h2>Single closed</h2>
    <OkuAccordion type="single" class="accordion-root">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Single open</h2>
    <OkuAccordion type="single" class="accordion-root" default-value="Two">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Multiple closed</h2>
    <OkuAccordion type="multiple" class="accordion-root">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Multiple open</h2>
    <OkuAccordion type="multiple" class="accordion-root" :default-value="['One', 'Two']">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h1>Controlled</h1>
    <h2>Single open</h2>
    <OkuAccordion type="single" class="accordion-root" value="Three">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Multiple open</h2>
    <OkuAccordion type="multiple" class="accordion-root" :value="['Two', 'Three']">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h1>Disabled (whole)</h1>
    <OkuAccordion type="single" class="accordion-root" disabled>
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h1>Disabled (item)</h1>
    <h2>Just item</h2>
    <OkuAccordion type="single" class="accordion-root">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item" :disabled="item === 'Two'">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>with `disabled=false` on top-level</h2>
    <OkuAccordion type="single" class="accordion-root" :disabled="false">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item" :disabled="item === 'Two'">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h1>Force mounted contents</h1>
    <OkuAccordion type="single" class="accordion-root">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content" force-mount>
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h1>State attributes</h1>
    <h2>Accordion disabled</h2>
    <OkuAccordion type="single" class="accordion-root-attr" default-value="Two" disabled>
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item-attr" :value="item">
        <OkuAccordionHeader class="accordion-header-attr">
          <OkuAccordionTrigger class="accordion-trigger-attr">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content-attr">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Accordion enabled with item override</h2>
    <OkuAccordion type="single" class="accordion-root-attr" default-value="Two" :disabled="false">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item-attr" :value="item" :disabled="['Two', 'Four'].includes(item)">
        <OkuAccordionHeader class="accordion-header-attr">
          <OkuAccordionTrigger class="accordion-trigger-attr">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content-attr">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Accordion disabled with item override</h2>
    <OkuAccordion type="single" class="accordion-root-attr" default-value="Two" :disabled="true">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item-attr" :value="item" :disabled="['Two', 'Four'].includes(item) ? false : undefined">
        <OkuAccordionHeader class="accordion-header-attr">
          <OkuAccordionTrigger class="accordion-trigger-attr">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content-attr">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
  </div>
</template>
