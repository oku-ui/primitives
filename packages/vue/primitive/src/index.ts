export {
  OkuPrimitive,
  Primitive,
} from './primitive'
export type { ComponentPropsWithoutRef, ElementRef } from './primitive'
