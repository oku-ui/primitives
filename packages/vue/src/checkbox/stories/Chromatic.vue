<script setup lang="ts">
import { OkuCheckbox, OkuCheckboxIndicator } from '@oku-ui/checkbox'
</script>

<template>
  <h1>Uncontrolled</h1>
  <h2>Unchecked</h2>
  <OkuCheckbox class="checkbox">
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h2>Checked</h2>
  <OkuCheckbox class="checkbox" default-checked>
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h1>Controlled</h1>
  <h2>Unchecked</h2>
  <OkuCheckbox class="checkbox" :checked="false">
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h2>Checked</h2>
  <OkuCheckbox class="checkbox" checked>
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h1>Indeterminate</h1>
  <OkuCheckbox class="checkbox" checked="indeterminate">
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h1>Disabled</h1>
  <OkuCheckbox class="checkbox" default-checked disabled>
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h1>Force mounted indicator</h1>
  <OkuCheckbox class="checkbox">
    <OkuCheckboxIndicator class="checkbox-indicator" force-mount :style="{ height: '20px' }" />
  </OkuCheckbox>

  <h1>State attributes</h1>
  <h2>Unchecked</h2>
  <OkuCheckbox class="checkbox-attr-styles">
    <OkuCheckboxIndicator class="checkbox-attr-styles" />
  </OkuCheckbox>

  <h2>Checked</h2>
  <OkuCheckbox class="checkbox-attr-styles" default-checked>
    <OkuCheckboxIndicator class="checkbox-attr-styles" />
  </OkuCheckbox>

  <h2>Indeterminate</h2>
  <OkuCheckbox class="checkbox-attr-styles" checked="indeterminate">
    <OkuCheckboxIndicator class="checkbox-attr-styles" />
  </OkuCheckbox>

  <h2>Disabled</h2>
  <OkuCheckbox class="checkbox-attr-styles" default-checked disabled>
    <OkuCheckboxIndicator class="checkbox-attr-styles" />
  </OkuCheckbox>

  <h2>Force mounted indicator</h2>
  <OkuCheckbox class="checkbox-attr-styles">
    <OkuCheckboxIndicator class="checkbox-attr-styles" force-mount :style="{ height: '20px' }" />
  </OkuCheckbox>
</template>
