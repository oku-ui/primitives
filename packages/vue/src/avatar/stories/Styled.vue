<script setup lang="ts">
import { OkuAvatar, OkuAvatarFallback, OkuAvatarImage } from '@oku-ui/avatar'
import AvatarIcon from './AvatarIcon.vue'

const src = 'https://picsum.photos/id/1005/400/400'
const srcBroken = 'https://broken.link.com/broken-pic.jpg'
</script>

<template>
  <h1>Without image & with fallback</h1>
  <OkuAvatar class="avatar">
    <OkuAvatarFallback class="avatar-fallback">
      JS
    </OkuAvatarFallback>
  </OkuAvatar>

  <h1>With image & with fallback</h1>
  <OkuAvatar class="avatar">
    <OkuAvatarImage class="avatar-image" alt="John Smith" :src="src" />
    <OkuAvatarFallback :delay-ms="300" class="avatar-fallback">
      JS
    </OkuAvatarFallback>
  </OkuAvatar>

  <h1>With image & with fallback (but broken src)</h1>
  <OkuAvatar class="avatar">
    <OkuAvatarImage
      class="avatar-image"
      alt="John Smith"
      :src="srcBroken"
      @loading-status-change="console.log($event)"
    />
    <OkuAvatarFallback class="avatar-fallback">
      <AvatarIcon />
    </OkuAvatarFallback>
  </OkuAvatar>
</template>
