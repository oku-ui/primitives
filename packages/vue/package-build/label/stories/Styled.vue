<script setup lang="ts">
import { OkuLabel } from '@oku-ui/label'
</script>

<template>
  <OkuLabel class="label">
    Label
  </OkuLabel>
</template>
