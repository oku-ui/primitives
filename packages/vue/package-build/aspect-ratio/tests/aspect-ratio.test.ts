import { enableAutoUnmount, shallowMount } from '@vue/test-utils'
import type { VueWrapper } from '@vue/test-utils'
import { afterEach, beforeEach, describe, expect, it } from 'vitest'
import { axe } from 'vitest-axe'

import { OkuAspectRatio } from '../src'

import Styled from '../src/stories/Styled.vue'
import CustomRatios from '../src/stories/CustomRatios.vue'
import Chromatic from '../src/stories/Chromatic.vue'

enableAutoUnmount(afterEach)

const RATIO = 1 / 2

describe('okuAspectRatio', () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = shallowMount({
      components: {
        OkuAspectRatio,
      },
      setup() {
        return {
          RATIO,
        }
      },
      template: `
        <div :style="{ width: '500px' }">
          <OkuAspectRatio :ratio="RATIO">
            <span>Hello</span>
          </OkuAspectRatio>
        </div>
      `,
    }, {
      attachTo: document.body,
    })
  })

  it('should render OkuAspectRatio correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()

    expect(shallowMount(OkuAspectRatio).html()).toMatchSnapshot()
  })

  /**
   * @vitest-environment jsdom
   */
  it('should have no accessibility violations', async () => {
    expect(await axe(wrapper.element)).toHaveNoViolations()
  })
})

describe('okuAspectRatio Stories', () => {
  describe('styled', () => {
    let wrapper: VueWrapper<InstanceType<typeof Styled>>

    beforeEach(async () => {
      wrapper = shallowMount(Styled, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('customRatios', () => {
    let wrapper: VueWrapper<InstanceType<typeof CustomRatios>>

    beforeEach(async () => {
      wrapper = shallowMount(CustomRatios, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('chromatic', () => {
    let wrapper: VueWrapper<InstanceType<typeof Chromatic>>

    beforeEach(async () => {
      wrapper = shallowMount(Chromatic, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
