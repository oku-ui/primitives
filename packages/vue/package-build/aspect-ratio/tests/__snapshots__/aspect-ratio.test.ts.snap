// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`okuAspectRatio > should render OkuAspectRatio correctly 1`] = `
"<div style="width: 500px;">
  <oku-aspect-ratio-stub ratio="0.5" aschild="false"></oku-aspect-ratio-stub>
</div>"
`;

exports[`okuAspectRatio > should render OkuAspectRatio correctly 2`] = `"<primitivediv-stub aschild="false" style="position: absolute; top: 0px; right: 0px; left: 0px; bottom: 0px;"></primitivediv-stub>"`;

exports[`okuAspectRatio Stories > chromatic > should render correctly 1`] = `
"<h1>Default ratio</h1>
<div style="width: 300px;">
  <oku-aspect-ratio-stub ratio="1" aschild="false" class="aspect-ratio"></oku-aspect-ratio-stub>
</div>
<h1>Custom ratios</h1>
<div style="display: flex; gap: 20px;">
  <div style="width: 200px;">
    <oku-aspect-ratio-stub ratio="0.5" aschild="false"></oku-aspect-ratio-stub>
  </div>
  <div style="width: 200px;">
    <oku-aspect-ratio-stub ratio="1" aschild="false"></oku-aspect-ratio-stub>
  </div>
  <div style="width: 200px;">
    <oku-aspect-ratio-stub ratio="1.7777777777777777" aschild="false"></oku-aspect-ratio-stub>
  </div>
  <div style="width: 200px;">
    <oku-aspect-ratio-stub ratio="2" aschild="false"></oku-aspect-ratio-stub>
  </div>
</div>"
`;

exports[`okuAspectRatio Stories > customRatios > should render correctly 1`] = `
"<div style="display: flex; gap: 20px;">
  <div style="width: 200px;">
    <oku-aspect-ratio-stub ratio="0.5" aschild="false"></oku-aspect-ratio-stub>
  </div>
  <div style="width: 200px;">
    <oku-aspect-ratio-stub ratio="1" aschild="false"></oku-aspect-ratio-stub>
  </div>
  <div style="width: 200px;">
    <oku-aspect-ratio-stub ratio="1.7777777777777777" aschild="false"></oku-aspect-ratio-stub>
  </div>
  <div style="width: 200px;">
    <oku-aspect-ratio-stub ratio="2" aschild="false"></oku-aspect-ratio-stub>
  </div>
</div>"
`;

exports[`okuAspectRatio Stories > styled > should render correctly 1`] = `
"<div style="width: 500px;">
  <oku-aspect-ratio-stub ratio="1" aschild="false" class="aspect-ratio"></oku-aspect-ratio-stub>
</div>"
`;
